tests/e2e/mod.rs

```diff
@@ -85,8 +94,36 @@ fn run_subgraphs_retail_supergraph() -> TempDir {
         })
         .expect("Could not execute connectivity check");
     }
-    // Return the folder the subgraphs are in
-    cloned_dir
+}
+
+#[fixture]
+#[once]
+fn reduced_supergraph_config(working_dir: &'static TempDir) -> ReducedSupergraphConfig {
+    println!("Cloning required git repository");
+    Repository::clone(
+        "https://github.com/apollosolutions/retail-supergraph",
+        working_dir.path(),
+    )
+    .expect("Could not clone supergraph repository");
+
+    // Jump into that temporary folder and run npm commands to kick off subgraphs
+    info!("Installing subgraph dependencies");
+    cmd!("npm", "install")
+        .dir(working_dir.path())
+        .run()
+        .expect("Could not install subgraph dependencies");
+
+    let supergraph_yaml_path =
+        Utf8PathBuf::from_path_buf(working_dir.path().join("supergraph-config-dev.yaml"))
+            .expect("Could not create path to config");
+
+    let content = std::fs::read_to_string(supergraph_yaml_path)
+        .expect("Could not read supergraph schema file");
+
+    let supergraph_config: ReducedSupergraphConfig =
+        serde_yaml::from_str(&content).expect("Could not parse supergraph schema file");
+
+    supergraph_config
```

Comment:
There's also now some duplication here as well, I feel like we could have two fixtures:

- `clone_retail_supergraph() -> ReducedSupergraphConfig`
- `run_retail_supergraph() -> ReducedSupergraphConfig` 

We could then further expand the ReducedSupergraphConfig to include the TempDir as a member of the struct so that it gets dropped when the ReducedSupergraphConfig gets dropped and that makes it a bit cleaner?

Reply:
Plus it stops us having to use fixtures that just return `()` and we might even want to rename ReducedSupergraphConfig to show it's something different...

Maybe a TestingSupergraphConfig or something similar?